tools/resourcedocsgen/cmd/docs/registry.go

```diff
@@ -39,6 +40,76 @@ import (
 	concpool "github.com/sourcegraph/conc/pool"
 )
 
+// PackageMetadataProvider is an interface for providers that retrieve package metadata
+// from either the filesystem directory or the Pulumi Registry API.
+type PackageMetadataProvider interface {
```

Comment:
Both `GetPackageMetadata` and `ListPackageMetadata` should take and use a `context.Context` for network operations.